mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -0,0 +1,146 @@
+OUTPUT_PATH = Path("~/github/mapping_parenting_tech/outputs/data/")
+STORE_PATH = "play_store/"
+PLAY_DATA_FILE = "playstore_innerHTML.html"
+OUTPUT_PLAY_IDS_FILE = "play_store_ids.json"
+OUTPUT_PLAY_DETAILS_FILE = "play_store_details.json"
+OUTPUT_PLAY_REVIEWS_FILE = "play_store_reviews.json"
+
+input_data = INPUT_PATH.expanduser() / STORE_PATH / PLAY_DATA_FILE
+
+if input_data.exists():
+    print("Ok to proceed")
+else:
+    print("Didn't find file")
+
+
+# %% [markdown]
+# ## For Play Store
+# Read HTML and extract list of app ids
+
+# %%
+html_doc = open(input_data, "r")
+html = html_doc.read()
+html_doc.close
+
+link_targets = re.findall(r"(?<=href=\"\/store\/apps\/details\?id=)(.*?)(?=\")", html)
+app_ids = list(dict.fromkeys(link_targets))
+output_ids = {"Parenting apps": app_ids}
+print("Apps ids parsed ok")
+
+# %% [markdown]
+# Save app ids to JSON file
+
+# %%
+output_target = OUTPUT_PATH.expanduser() / OUTPUT_PLAY_IDS_FILE
+with open(output_target, "w") as output_file:
+    json.dump(output_ids, output_file)
+output_file.close
+print("App ids saved to", output_target)
+
+# %% [markdown]
+# Retrieve app details
+
+# %%
+all_app_details = dict()
+remove_apps = list()
+review_counter = 0
+most_reviews = {"id": 0, "count": 0}
+
+for app_id in tqdm(output_ids["Parenting apps"], desc="Retrieving app details"):
+    try:
+        app_details = app(
+            app_id, lang="en", country="gb"  # defaults to 'en'  # defaults to 'us'
+        )
+        all_app_details.update({app_id: app_details})
+        review_counter += app_details["reviews"]
+        if app_details["reviews"] > most_reviews["count"]:
+            most_reviews["id"] = app_id
+            most_reviews["count"] = app_details["reviews"]
+
+    except Exception as e:
+        print(f"Error on app id {app_id}: {e} {repr(e)}")
+        remove_apps.append(app_id)
+
+for app_id in remove_apps:
+    output_ids["Parenting apps"].remove(app_id)
+
+print(
+    f"There are {review_counter} reviews available for {len(output_ids['Parenting apps'])} apps. {len(remove_apps)} apps were removed"
+)
+print(f"App id {most_reviews['id']} had the most reviews with {most_reviews['count']}")
+
+# %% [markdown]
+# Save app details
+
+# %%
+output_target = OUTPUT_PATH.expanduser() / OUTPUT_PLAY_DETAILS_FILE
+with open(output_target, "w") as output_file:
+    json.dump(all_app_details, output_file, indent=2, default=str)
+output_file.close
+
+print(f"{len(all_app_details)} apps' details saved to {output_target}")
+
+# %% [markdown]
+# Download reviews for the apps in the list
+
+# %%
+all_app_reviews = dict()
+running_total = 0
+
+for app_id in tqdm(output_ids["Parenting apps"], desc="Retrieving app reviews"):
+    try:
+        app_reviews = reviews_all(
+            app_id,
+            sleep_milliseconds=0,  # defaults to 0
+            lang="en",
+            country="gb",
+            sort=Sort.MOST_RELEVANT  # defaults to Sort.MOST_RELEVANT
```

Comment:
Leave this to you, but perhaps it's better to sort by date, if possible? (that seems to be a more stable sort order than relevancy which might be stable, but might also change from run to run if play store updates the relevancy scores...)

Reply:
Yes, will do.